src/Context/FixtureContext.php

```diff
@@ -699,11 +698,29 @@ public function stepGoToNamedRecord($type, $id)
      */
     public function iAddAnExtensionToTheClass($extension, $class)
     {
+        $this->iAddAnExtensionToTheClassInner($extension, $class, false);
+    }
+
+    /**
+     * @param $extension
+     * @param $class
+     *
+     * @Given I add an extension :extension to the :class class and flush only
+     */
+    public function iAddAnExtensionToTheClassAndFlushOnly(string $extension, string $class): void
+    {
+        $this->iAddAnExtensionToTheClassInner($extension, $class, true);
+    }
+
+    private function iAddAnExtensionToTheClassInner(string $extension, string $class, bool $flushOnly)
+    {
```

Comment:
What's this distinction for? And what does "flush only" mean? That seems to imply _something_ else isn't happening? But it isn't clear what.

Reply:
On line 763 of this class it will either `dev/build?flush` or simply `?flush`.

The reason you might not want a full dev/build after adding the extension is that you don't want the default pages created (home, about us, etc) - doing that will break this test https://github.com/silverstripe/silverstripe-cms/pull/2992/files#diff-72a04b0ebd4622033a8873689c3beb1421a45eadd93c84632098d1668f734d4aR8 as it uses an 'About us' page and asserts against an ID of 2.

Prior to all these PRs there was no dev/build happening because the extension being added in that test extended Extension rather than DataExtension, and the old logic of this method meant it would flush only. Now that we don't have that DataExtension logic any more we need a new way to say 'flush only'

Reply:
Can we please update the text so it's clearer when looking at the feature files what's happening?
Maybe something like this:

> @Given I add an extension :extension to the :class class without dev/build

It's more important in these cases what _isn't_ happening, rather than what _is_ happening. Otherwise in a year's time when we're adding a new extension to some feature we'll see "flush only" and have to look back to see what that's actually excluding.

Alternatively it might be better to flip this on its head - have flush only be the default when you use 

> @Given I add an extension :extension to the :class class

And then the new one can be 

> @Given I add an extension :extension to the :class class and run dev/build

That way we only run dev/build when we intentionally want to, rather than just 'cause we forgot to use the "without dev/build" variant.

Reply:
I'm not fussed which of those you go with, so long as the language is clearly about whether or not `dev/build` gets run rather than "only flush'

Reply:
Have gone with option 1